File: herencia-dto/AMarotoChivite-HerenciaDTO/AMaroto-Chivite-HerenciasDto-Java/src/main/java/Ficheros/HerenciasDto/models/factories/PersonFactory.java
package Ficheros.HerenciasDto.models.factories;

import Ficheros.HerenciasDto.models.dto.PersonDTO;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public class PersonFactory {
    private final List<String> modulos = Arrays.asList("Programación", "Entornos");
    private final List<String> nombres = Arrays.asList(
            "Juan", "Ana", "Pedro", "María", "Luis", "Carla", "David", "Elena",
            "Fernando", "Sara", "Javier", "Lucía", "Miguel", "Pilar", "Rosa", "Sofía",
            "Diego", "Jorge", "Laura", "Raquel"
    );

    private PersonDTO createRandomPersona() {
        double random = Math.random();
        String name = nombres.get(new Random().nextInt(nombres.size()));
        int edad = new Random().nextInt(23) + 18;

        if (random < 0.1) {
            String modulo = modulos.get(new Random().nextInt(modulos.size()));
            return new PersonDTO.Profesor(name, Integer.toString(edad), modulo);
        } else {
            return new PersonDTO.Alumno(name, Integer.toString(edad));
        }
    }

    public List<PersonDTO> createPersonas(int n) {
        List<PersonDTO> personas = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            personas.add(createRandomPersona());
        }
        return personas;
    }
}

File: herencia-dto/AMarotoChivite-HerenciaDTO/AMarotoChivite-HerenciasDto-Kotlin/src/main/kotlin/Ficheros/HerenciasDto/models/factories/PersonFactory.kt
package Ficheros.HerenciasDto.models.factories

import Ficheros.HerenciasDto.models.dto.PersonDTO
import kotlin.random.Random

class PersonaFactory {
    private val modulos = listOf("Programación", "Entornos")
    private val nombres = listOf(
        "Juan", "Ana", "Pedro", "María", "Luis", "Carla", "David", "Elena",
        "Fernando", "Sara", "Javier", "Lucía", "Miguel", "Pilar", "Rosa", "Sofía",
        "Diego", "Jorge", "Laura", "Raquel"
    )

    private fun createRandomPersona(): PersonDTO {
        val random = Random.nextDouble()
        val name = nombres.random()
        val edad = (18..40).random()

        return if (random < 0.1) {
            val modulo = modulos.random()
            PersonDTO.Profesor(name, edad.toString(), modulo)
        } else {
            PersonDTO.Alumno(name, edad.toString())
        }
    }

    fun createPersonas(n: Int): List<PersonDTO> {
        val personas = mutableListOf<PersonDTO>()
        repeat(n) {
            personas.add(createRandomPersona())
        }
        return personas
    }
}